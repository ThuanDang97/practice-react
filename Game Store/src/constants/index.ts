export * from '@constants/errorMessage'
export * from '@constants/regex'
export * from '@constants/endPoints'
