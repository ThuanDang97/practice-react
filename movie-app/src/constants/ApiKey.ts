const API_KEY = '774dadd71369f920d925c1ad58e2bbbe';

const API_READ_ACCESS_TOKEN =
  'eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI3NzRkYWRkNzEzNjlmOTIwZDkyNWMxYWQ1OGUyYmJiZSIsInN1YiI6IjYyODFhOTk0NmQwMDBjMDA5Y2M5OWExMSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.XQvBkWOHmC_5hKWCrLKIKV-_oEH3SaDg6nICpUegh9A';

const SESSION_ID = '4c5425ed4c8d603768ab3a50124c5083a12f09d1';

const USER_ID = '12410501';

export { API_KEY, API_READ_ACCESS_TOKEN, SESSION_ID, USER_ID };
