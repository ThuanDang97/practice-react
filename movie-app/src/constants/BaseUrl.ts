const BASE_URL = 'https://api.themoviedb.org/3';

export { BASE_URL };
