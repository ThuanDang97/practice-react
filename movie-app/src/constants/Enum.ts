enum DefaultRangerValue {
  Min = 0,
  Max = 10,
  DefaultValue = 10,
  Step = 1,
}

enum MovieType {
  NowPlaying = 'now-playing',
  UpComing = 'up-coming',
  TopRated = 'top-rated',
  Favorite = 'favorite-movies',
  Popular = 'popular',
}

export { DefaultRangerValue, MovieType };
