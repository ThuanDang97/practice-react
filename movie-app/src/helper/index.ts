import useSWR from 'swr';
import { IMovie } from '../interfaces/Movie/IMovie';
import { IMovieCard } from '../interfaces/Movie/IMovieCard';
import { IMovieInfo } from '../interfaces/Movie/IMovieInfo';

export type TResult = IMovieCard | IMovieInfo | IMovie;

const transformedData = (result1: TResult[], result2: TResult[]) => {
  if (!result1 || !result2) {
    return [];
  }

  const newDataTransform = result1.map((item) => {
    return {
      ...item,
      is_favorite: !!result2.find((favorite) => item.id === favorite.id),
    };
  });

  return newDataTransform;
};

const fetcher = async (url: string) => {
  const response = await fetch(url);
  const responseData = await response.json();

  if (response.ok) {
    return responseData;
  }
};

const fetcherAll = (...urls: string[]) => {
  return Promise.all(urls.map((url) => fetcher(url)));
};

const fetchAPI = (url: string) => {
  const { data, error } = useSWR(url, fetcher);

  if (error) {
    return 'failed to load';
  }
  return data;
};

const fetchMultiAPI = (urls: string[]) => {
  const { data, error } = useSWR(urls, fetcherAll);

  if (!error) {
    return data;
  }
};

const convertMinsToHrsMins = (minutes: number) => {
  let hour = Math.floor(minutes / 60);
  let minute = minutes % 60;

  return hour + 'h ' + minute + 'm';
};

export {
  convertMinsToHrsMins,
  fetchAPI,
  fetchMultiAPI,
  transformedData,
  fetcher,
  fetcherAll,
};
