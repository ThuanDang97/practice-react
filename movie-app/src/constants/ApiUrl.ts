import { API_KEY, SESSION_ID, USER_ID } from './ApiKey';
import { BASE_URL } from './BaseUrl';

const MOVIE_POPULAR_URL = `${BASE_URL}/discover/movie?api_key=${API_KEY}&sort_by=popularity.desc&page=&page=1`;

const MOVIE_NOW_PLAYING_URL = `${BASE_URL}/movie/now_playing?api_key=${API_KEY}&language=en-US&page=1`;

const MOVIE_UPCOMING_URL = `${BASE_URL}/movie/upcoming?api_key=${API_KEY}&language=en-US&page=1`;

const MOVIE_TOP_RATED_URL = `${BASE_URL}/movie/top_rated?api_key=${API_KEY}&language=en-US&page=1`;

const MOVIE_FAVORITE_URL = `${BASE_URL}/account/${USER_ID}/favorite/movies?api_key=${API_KEY}&session_id=${SESSION_ID}&language=en-US&sort_by=created_at.asc&page=1`;

const UPDATE_FAVORITE_URL = `${BASE_URL}/account/${USER_ID}/favorite?api_key=${API_KEY}&session_id=${SESSION_ID}`;

const GENRES_URL = `${BASE_URL}/genre/movie/list?api_key=${API_KEY}&language=en-US`;

export {
  MOVIE_POPULAR_URL,
  MOVIE_NOW_PLAYING_URL,
  MOVIE_UPCOMING_URL,
  MOVIE_TOP_RATED_URL,
  MOVIE_FAVORITE_URL,
  UPDATE_FAVORITE_URL,
  GENRES_URL,
};
