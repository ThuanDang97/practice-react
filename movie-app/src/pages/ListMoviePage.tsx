import { ChangeEvent, MouseEvent, useCallback, useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import ListMovie from '../components/ListMovie/ListMovie';
import {
  GENRES_URL,
  MOVIE_FAVORITE_URL,
  MOVIE_NOW_PLAYING_URL,
  MOVIE_POPULAR_URL,
  MOVIE_TOP_RATED_URL,
  MOVIE_UPCOMING_URL,
} from '../constants/ApiUrl';
import { MovieType } from '../constants/Enum';
import { fetchAPI, fetchMultiAPI, transformedData } from '../helper';
import { IChip } from '../interfaces/Chip/IChip';
import { IMovieCard } from '../interfaces/Movie/IMovieCard';
import { updateFavorite, getTitlePage, getNewMovie, getNewChip } from '../utils';

const ListMovieType = (movieType: MovieType) => {
  switch (movieType) {
    case MovieType.NowPlaying:
      return MOVIE_NOW_PLAYING_URL;
    case MovieType.UpComing:
      return MOVIE_UPCOMING_URL;
    case MovieType.TopRated:
      return MOVIE_TOP_RATED_URL;
    case MovieType.Favorite:
      return MOVIE_FAVORITE_URL;
    default:
      return MOVIE_POPULAR_URL;
  }
};

const ListMoviePage = (): JSX.Element => {
  const { movieType = MovieType.Popular } = useParams() as { movieType: MovieType };
  let url = ListMovieType(movieType as MovieType);
  const movieOriginal = fetchMultiAPI([url, MOVIE_FAVORITE_URL]);
  const originalChips = fetchAPI(GENRES_URL);
  const [movies, setMovies] = useState<IMovieCard[]>([]);
  const [chips, setChips] = useState<IChip[]>([]);
  const [totalPage, setTotalPage] = useState<number>(1);
  const [sort, setSort] = useState<string>('popularity.desc');
  const [genres, setGenres] = useState<number[]>([]);
  const [score, setScore] = useState<number>(10);
  const [count, setCount] = useState<number>(1);
  let isDisableLoadMore = false;
  let isDisableSearch = false;

  useEffect(() => {
    if (movieOriginal) {
      const newMovie = transformedData(
        movieOriginal[0].results,
        movieOriginal[1].results,
      );

      setMovies(newMovie);
      setTotalPage(movieOriginal[0].total_pages);
    }

    if (originalChips) {
      const newChips: IChip[] = originalChips.genres.map((chip: any) => {
        return { ...chip, isChip: false };
      });

      setChips(newChips);
    }
  }, [movieOriginal, movieType, originalChips]);

  const handleUpdateStatus = useCallback(
    async (movie: IMovieCard) => {
      const status = await updateFavorite(movie.id, movie.is_favorite);

      if (movieType === 'favorite-movies') {
        const newMovies = movies?.map((PrevMovie) => {
          return PrevMovie.id === movie.id
            ? { ...PrevMovie, is_favorite: !PrevMovie.is_favorite }
            : PrevMovie;
        });

        if (status === 'success' && newMovies) {
          setMovies(
            newMovies.filter((movie) => {
              return movie.is_favorite === true;
            }),
          );
        }
      } else {
        setMovies(
          movies.map((PrevMovie) => {
            return PrevMovie.id === movie.id
              ? { ...PrevMovie, is_favorite: !PrevMovie.is_favorite }
              : PrevMovie;
          }),
        );
      }
    },
    [movieType, movies],
  );

  const handleLoadMore = async () => {
    const page = count + 1;

    const newMovie = await getNewMovie({
      movieType,
      count: page,
      genres,
      score,
      sort,
    });

    if (newMovie) {
      setMovies([...movies, ...newMovie]);
      setCount(page);
    }
  };

  // TODO:
  // - The API only allows sorting of popular movies
  // NEED:
  // - Sort all pages movies

  const handleSortMovie = async (option: string) => {
    const newMovie = await getNewMovie({
      movieType,
      count,
      genres,
      score,
      sort: option,
    });

    if (newMovie) {
      setMovies([...newMovie]);
      setSort(option);
      setCount(1);
    }
  };

  const handleFilterGenres = useCallback(
    (e: MouseEvent<HTMLButtonElement>) => {
      const idGenres = Number(e.currentTarget.value);
      const isExist = genres.find((gen) => gen === idGenres);

      if (isExist) {
        const newGenres = genres.filter((gen) => gen !== idGenres);
        setGenres(newGenres);
      } else {
        setGenres([...genres, idGenres]);
      }

      const newChips = getNewChip(chips, idGenres);
      setChips(newChips);
    },
    [chips, genres],
  );

  const handleFilterUserScore = (e: ChangeEvent<HTMLInputElement>) => {
    const value = Number(e.target.value);
    setScore(value);
  };

  const handleSearchMovie = async () => {
    const newMovie = await getNewMovie({
      movieType,
      count,
      genres,
      score,
      sort,
    });

    if (newMovie) {
      setMovies([...newMovie]);
      setCount(1);
    }
  };

  const onDisableLoadMore =
    count === totalPage || movies?.length === 0 ? !isDisableLoadMore : isDisableLoadMore;

  const title = getTitlePage(movieType as MovieType);

  return (
    <ListMovie
      pageMovie={movies}
      title={title}
      chips={chips}
      onUpdateStatus={handleUpdateStatus}
      onDisableLoadMore={onDisableLoadMore}
      onDisableSearch={isDisableSearch}
      onHandleClick={handleLoadMore}
      onHandleSort={handleSortMovie}
      onHandleFilterGenres={handleFilterGenres}
      onHandleFilterUserScore={handleFilterUserScore}
      onHandleSearch={handleSearchMovie}
    />
  );
};
export default ListMoviePage;
