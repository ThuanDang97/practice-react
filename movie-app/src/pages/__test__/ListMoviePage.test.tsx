import { screen, render, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { API_KEY, USER_ID, SESSION_ID } from '../../constants/ApiKey';
import { BASE_URL } from '../../constants/BaseUrl';
import { MovieType } from '../../constants/Enum';
import ListMoviePage, { ListMovieType } from '../ListMoviePage';
import {
  mockResult,
  mockMovieOriginal,
  mockChipOriginal,
} from '../../constants/MockData';
import * as helpers from '../../helper';
import { setup } from '../../support/Setup';

const { fetchMultiAPI, fetchAPI } = helpers;

jest.mock('../../helper');
const mockTransformedData = jest.spyOn(helpers, 'transformedData');
describe('Test List Movie Page', () => {
  beforeEach(() => {
    jest.mock('react-router-dom', () => ({
      ...jest.requireActual('react-router-dom'),
      useParams: jest.fn(() => ({
        movieType: 'popular',
      })),
      useRouteMatch: () => ({ url: '' }),
    }));

    (fetchMultiAPI as jest.Mock).mockImplementation(() => mockMovieOriginal);
    mockTransformedData.mockImplementation(() => mockResult);
    (fetchAPI as jest.Mock).mockImplementation(() => mockChipOriginal);
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  const createWrapper = () => {
    return render(
      <Router>
        <ListMoviePage />
      </Router>,
    );
  };
  test('List Movie Page matches DOM Snapshot', () => {
    const utils = createWrapper();

    expect(utils).toMatchSnapshot();
  });

  test('List Movie Page fetching data successfully', () => {
    createWrapper();

    expect(screen.getByTestId('header-page')).toHaveTextContent('Popular Movies');
    expect(screen.getByTestId('list-chip')).toBeInTheDocument();
    expect(screen.queryByTestId('message-error')).not.toBeInTheDocument();
  });

  test('Function handleFilterUserScore should return invalid value', () => {
    createWrapper();

    fireEvent.change(screen.getByRole('slider'), { target: { value: 9 } });
    expect(screen.getByRole('slider')).toHaveValue(String(9));
  });

  test('Function handleLoadMore should return invalid value', async () => {
    const { user } = setup(
      <Router>
        <ListMoviePage />
      </Router>,
    );

    await user.click(screen.getByRole('button', { name: /Load More/i }));
  });
  test('Function ListMovieType should be return url correctly', () => {
    expect(ListMovieType(MovieType.Popular)).toBe(
      `${BASE_URL}/discover/movie?api_key=${API_KEY}&sort_by=popularity.desc&page=&page=1`,
    );
    expect(ListMovieType(MovieType.NowPlaying)).toBe(
      `${BASE_URL}/movie/now_playing?api_key=${API_KEY}&language=en-US&page=1`,
    );
    expect(ListMovieType(MovieType.UpComing)).toBe(
      `${BASE_URL}/movie/upcoming?api_key=${API_KEY}&language=en-US&page=1`,
    );
    expect(ListMovieType(MovieType.TopRated)).toBe(
      `${BASE_URL}/movie/top_rated?api_key=${API_KEY}&language=en-US&page=1`,
    );
    expect(ListMovieType(MovieType.Favorite)).toBe(
      `${BASE_URL}/account/${USER_ID}/favorite/movies?api_key=${API_KEY}&session_id=${SESSION_ID}&language=en-US&sort_by=created_at.asc&page=1`,
    );
  });
});
