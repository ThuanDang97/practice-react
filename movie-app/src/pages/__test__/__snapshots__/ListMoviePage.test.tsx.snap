// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test List Movie Page List Movie Page matches DOM Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="media"
        data-testid="home-page"
      >
        <div
          class="column_wrapper"
        >
          <div
            class="content_wrapper"
          >
            <div
              class="title"
            >
              <h2
                data-testid="header-page"
              >
                Popular Movies
              </h2>
            </div>
            <div
              class="content"
            >
              <div>
                <div
                  class="panel-card"
                >
                  <div
                    class="name"
                  >
                    <h2>
                      Sort
                    </h2>
                  </div>
                  <div
                    class="filter"
                  >
                    <span
                      class="widget"
                      data-testid="sub-menu"
                    >
                      <span
                        class="dropdown"
                        data-testid="select"
                      >
                        <span
                          class="input"
                          data-testid="titleOption"
                        >
                          Popularity Descending
                        </span>
                        <span
                          class="select"
                        >
                          <span
                            class="icon-select"
                          />
                        </span>
                      </span>
                    </span>
                  </div>
                </div>
                <div
                  class="panel-card"
                >
                  <div
                    class="name"
                  >
                    <h2>
                      Filter
                    </h2>
                  </div>
                  <div
                    class="filter"
                  >
                    <span
                      class="widget"
                      data-testid="sub-menu"
                    >
                      <div
                        class="list-chip"
                        data-testid="list-chip"
                      >
                        <h3>
                          Genres
                        </h3>
                        <button
                          class="chip-element"
                          value="28"
                        >
                          Action
                        </button>
                        <button
                          class="chip-element"
                          value="12"
                        >
                          Adventure
                        </button>
                        <button
                          class="chip-element"
                          value="16"
                        >
                          Animation
                        </button>
                        <button
                          class="chip-element"
                          value="35"
                        >
                          Comedy
                        </button>
                        <button
                          class="chip-element"
                          value="80"
                        >
                          Crime
                        </button>
                        <button
                          class="chip-element"
                          value="99"
                        >
                          Documentary
                        </button>
                        <button
                          class="chip-element"
                          value="18"
                        >
                          Drama
                        </button>
                        <button
                          class="chip-element"
                          value="10751"
                        >
                          Family
                        </button>
                        <button
                          class="chip-element"
                          value="14"
                        >
                          Fantasy
                        </button>
                        <button
                          class="chip-element"
                          value="36"
                        >
                          History
                        </button>
                        <button
                          class="chip-element"
                          value="27"
                        >
                          Horror
                        </button>
                        <button
                          class="chip-element"
                          value="10402"
                        >
                          Music
                        </button>
                        <button
                          class="chip-element"
                          value="9648"
                        >
                          Mystery
                        </button>
                        <button
                          class="chip-element"
                          value="10749"
                        >
                          Romance
                        </button>
                        <button
                          class="chip-element"
                          value="878"
                        >
                          Science Fiction
                        </button>
                        <button
                          class="chip-element"
                          value="10770"
                        >
                          TV Movie
                        </button>
                        <button
                          class="chip-element"
                          value="53"
                        >
                          Thriller
                        </button>
                        <button
                          class="chip-element"
                          value="10752"
                        >
                          War
                        </button>
                        <button
                          class="chip-element"
                          value="37"
                        >
                          Western
                        </button>
                      </div>
                      <div
                        class="ranger-slider"
                      >
                        <h3>
                          User Score
                        </h3>
                        <input
                          class="range"
                          max="10"
                          min="0"
                          step="1"
                          type="range"
                          value="10"
                        />
                        <div
                          class="range-value"
                        >
                          10
                        </div>
                      </div>
                    </span>
                  </div>
                </div>
                <div
                  class="btn-layout"
                >
                  <button
                    class="btn primary contained rd-2 w-100 "
                    type="button"
                  >
                    Search
                  </button>
                </div>
              </div>
              <div
                class="white_column"
              >
                <section
                  class="page_wrapper"
                >
                  <div
                    class="card"
                  >
                    <div
                      class="image"
                    >
                      <div
                        class="wrapper"
                      >
                        <a
                          href="/movie/979163"
                        >
                          <img
                            alt="Beyond Infinity: Buzz and the Journey to Lightyear"
                            class="poster"
                            src="https://www.themoviedb.org/t/p/w220_and_h330_face//3dhQksZrIeRYP8jbB0J1SLW6HVW.jpg"
                          />
                        </a>
                        <div
                          class="option"
                          data-testid="optional"
                        >
                          <span />
                        </div>
                      </div>
                    </div>
                    <div
                      class="content"
                    >
                      <div
                        class="consensus tight"
                      >
                        <div
                          class="outer_ring"
                        >
                          <div
                            class="user_score_chart"
                            data-bar-color="#21d07a"
                            data-percent="78.0"
                            data-track-color="#204529"
                          >
                            <div
                              class="percent"
                            >
                              <span
                                class="icon"
                                data-testid="score"
                              >
                                74
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                      <h2>
                        <a
                          href="/movie/979163"
                          title="Beyond Infinity: Buzz and the Journey to Lightyear"
                        >
                          Beyond Infinity: Buzz and the Journey to Lightyear
                        </a>
                      </h2>
                      <p
                        data-testid="release-date"
                      >
                        2022-06-10
                      </p>
                    </div>
                  </div>
                  <div
                    class="card"
                  >
                    <div
                      class="image"
                    >
                      <div
                        class="wrapper"
                      >
                        <a
                          href="/movie/453395"
                        >
                          <img
                            alt="Doctor Strange in the Multiverse of Madness"
                            class="poster"
                            src="https://www.themoviedb.org/t/p/w220_and_h330_face//3dhQksZrIeRYP8jbB0J1SLW6HVW.jpg"
                          />
                        </a>
                        <div
                          class="option"
                          data-testid="optional"
                        >
                          <span />
                        </div>
                      </div>
                    </div>
                    <div
                      class="content"
                    >
                      <div
                        class="consensus tight"
                      >
                        <div
                          class="outer_ring"
                        >
                          <div
                            class="user_score_chart"
                            data-bar-color="#21d07a"
                            data-percent="78.0"
                            data-track-color="#204529"
                          >
                            <div
                              class="percent"
                            >
                              <span
                                class="icon"
                                data-testid="score"
                              >
                                74
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                      <h2>
                        <a
                          href="/movie/453395"
                          title="Doctor Strange in the Multiverse of Madness"
                        >
                          Doctor Strange in the Multiverse of Madness
                        </a>
                      </h2>
                      <p
                        data-testid="release-date"
                      >
                        2022-06-10
                      </p>
                    </div>
                  </div>
                </section>
                <div
                  class="btn-layout"
                >
                  <button
                    class="btn primary contained rd-1 w-100 "
                    type="button"
                  >
                    Load More
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="media"
      data-testid="home-page"
    >
      <div
        class="column_wrapper"
      >
        <div
          class="content_wrapper"
        >
          <div
            class="title"
          >
            <h2
              data-testid="header-page"
            >
              Popular Movies
            </h2>
          </div>
          <div
            class="content"
          >
            <div>
              <div
                class="panel-card"
              >
                <div
                  class="name"
                >
                  <h2>
                    Sort
                  </h2>
                </div>
                <div
                  class="filter"
                >
                  <span
                    class="widget"
                    data-testid="sub-menu"
                  >
                    <span
                      class="dropdown"
                      data-testid="select"
                    >
                      <span
                        class="input"
                        data-testid="titleOption"
                      >
                        Popularity Descending
                      </span>
                      <span
                        class="select"
                      >
                        <span
                          class="icon-select"
                        />
                      </span>
                    </span>
                  </span>
                </div>
              </div>
              <div
                class="panel-card"
              >
                <div
                  class="name"
                >
                  <h2>
                    Filter
                  </h2>
                </div>
                <div
                  class="filter"
                >
                  <span
                    class="widget"
                    data-testid="sub-menu"
                  >
                    <div
                      class="list-chip"
                      data-testid="list-chip"
                    >
                      <h3>
                        Genres
                      </h3>
                      <button
                        class="chip-element"
                        value="28"
                      >
                        Action
                      </button>
                      <button
                        class="chip-element"
                        value="12"
                      >
                        Adventure
                      </button>
                      <button
                        class="chip-element"
                        value="16"
                      >
                        Animation
                      </button>
                      <button
                        class="chip-element"
                        value="35"
                      >
                        Comedy
                      </button>
                      <button
                        class="chip-element"
                        value="80"
                      >
                        Crime
                      </button>
                      <button
                        class="chip-element"
                        value="99"
                      >
                        Documentary
                      </button>
                      <button
                        class="chip-element"
                        value="18"
                      >
                        Drama
                      </button>
                      <button
                        class="chip-element"
                        value="10751"
                      >
                        Family
                      </button>
                      <button
                        class="chip-element"
                        value="14"
                      >
                        Fantasy
                      </button>
                      <button
                        class="chip-element"
                        value="36"
                      >
                        History
                      </button>
                      <button
                        class="chip-element"
                        value="27"
                      >
                        Horror
                      </button>
                      <button
                        class="chip-element"
                        value="10402"
                      >
                        Music
                      </button>
                      <button
                        class="chip-element"
                        value="9648"
                      >
                        Mystery
                      </button>
                      <button
                        class="chip-element"
                        value="10749"
                      >
                        Romance
                      </button>
                      <button
                        class="chip-element"
                        value="878"
                      >
                        Science Fiction
                      </button>
                      <button
                        class="chip-element"
                        value="10770"
                      >
                        TV Movie
                      </button>
                      <button
                        class="chip-element"
                        value="53"
                      >
                        Thriller
                      </button>
                      <button
                        class="chip-element"
                        value="10752"
                      >
                        War
                      </button>
                      <button
                        class="chip-element"
                        value="37"
                      >
                        Western
                      </button>
                    </div>
                    <div
                      class="ranger-slider"
                    >
                      <h3>
                        User Score
                      </h3>
                      <input
                        class="range"
                        max="10"
                        min="0"
                        step="1"
                        type="range"
                        value="10"
                      />
                      <div
                        class="range-value"
                      >
                        10
                      </div>
                    </div>
                  </span>
                </div>
              </div>
              <div
                class="btn-layout"
              >
                <button
                  class="btn primary contained rd-2 w-100 "
                  type="button"
                >
                  Search
                </button>
              </div>
            </div>
            <div
              class="white_column"
            >
              <section
                class="page_wrapper"
              >
                <div
                  class="card"
                >
                  <div
                    class="image"
                  >
                    <div
                      class="wrapper"
                    >
                      <a
                        href="/movie/979163"
                      >
                        <img
                          alt="Beyond Infinity: Buzz and the Journey to Lightyear"
                          class="poster"
                          src="https://www.themoviedb.org/t/p/w220_and_h330_face//3dhQksZrIeRYP8jbB0J1SLW6HVW.jpg"
                        />
                      </a>
                      <div
                        class="option"
                        data-testid="optional"
                      >
                        <span />
                      </div>
                    </div>
                  </div>
                  <div
                    class="content"
                  >
                    <div
                      class="consensus tight"
                    >
                      <div
                        class="outer_ring"
                      >
                        <div
                          class="user_score_chart"
                          data-bar-color="#21d07a"
                          data-percent="78.0"
                          data-track-color="#204529"
                        >
                          <div
                            class="percent"
                          >
                            <span
                              class="icon"
                              data-testid="score"
                            >
                              74
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                    <h2>
                      <a
                        href="/movie/979163"
                        title="Beyond Infinity: Buzz and the Journey to Lightyear"
                      >
                        Beyond Infinity: Buzz and the Journey to Lightyear
                      </a>
                    </h2>
                    <p
                      data-testid="release-date"
                    >
                      2022-06-10
                    </p>
                  </div>
                </div>
                <div
                  class="card"
                >
                  <div
                    class="image"
                  >
                    <div
                      class="wrapper"
                    >
                      <a
                        href="/movie/453395"
                      >
                        <img
                          alt="Doctor Strange in the Multiverse of Madness"
                          class="poster"
                          src="https://www.themoviedb.org/t/p/w220_and_h330_face//3dhQksZrIeRYP8jbB0J1SLW6HVW.jpg"
                        />
                      </a>
                      <div
                        class="option"
                        data-testid="optional"
                      >
                        <span />
                      </div>
                    </div>
                  </div>
                  <div
                    class="content"
                  >
                    <div
                      class="consensus tight"
                    >
                      <div
                        class="outer_ring"
                      >
                        <div
                          class="user_score_chart"
                          data-bar-color="#21d07a"
                          data-percent="78.0"
                          data-track-color="#204529"
                        >
                          <div
                            class="percent"
                          >
                            <span
                              class="icon"
                              data-testid="score"
                            >
                              74
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                    <h2>
                      <a
                        href="/movie/453395"
                        title="Doctor Strange in the Multiverse of Madness"
                      >
                        Doctor Strange in the Multiverse of Madness
                      </a>
                    </h2>
                    <p
                      data-testid="release-date"
                    >
                      2022-06-10
                    </p>
                  </div>
                </div>
              </section>
              <div
                class="btn-layout"
              >
                <button
                  class="btn primary contained rd-1 w-100 "
                  type="button"
                >
                  Load More
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
